lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1611,13 +1616,33 @@ def _set_unit_ip(self, _):
         event is actually needed.
         """
         for relation in self._charm.model.relations[self._relation_name]:
-            relation.data[self._charm.unit]["prometheus_scrape_unit_address"] = str(
-                self._charm.model.get_binding(relation).network.bind_address
-            )
+            unit_ip = str(self._charm.model.get_binding(relation).network.bind_address)
+
+            if not self._is_valid_unit_address(unit_ip):
+                # relation data will be updated later when a valid address becomes available
+                continue
+
+            relation.data[self._charm.unit]["prometheus_scrape_unit_address"] = unit_ip
             relation.data[self._charm.unit]["prometheus_scrape_unit_name"] = str(
                 self._charm.model.unit.name
             )
 
+    def _is_valid_unit_address(self, address: str) -> bool:
```

Comment:
I think we can trust ops to deliver on their promise for `bind_address`: either None or an address.
I.e. I would drop this function and use `if bind_address` inline.

Reply:
I see no harm in actually validating the IP address.